units/provisioning_create_host.go

```diff
@@ -144,14 +145,50 @@ func (j *createHostJob) Run(ctx context.Context) {
 			grip.Error(message.WrapError(err, message.Fields{
 				"host_id":  j.HostID,
 				"attempt":  j.CurrentAttempt,
-				"distro":   j.host.Distro,
+				"distro":   j.host.Distro.Id,
 				"job":      j.ID(),
 				"provider": j.host.Provider,
 				"message":  "could not remove intent document",
 				"outcome":  "host pool may exceed maxhost limit",
 			}))
 
 			return
+		} else if !j.host.IsSubjectToHostCreationThrottle() {
+			// pass (spawn hosts and friends should always
+			// try and start)
+		} else if j.host.Status == evergreen.HostBuilding {
+			// pass (containers or similar that are
+			// building should continue to provision)
+		} else {
+			var (
+				numProv            int
+				runningHosts       int
+				distroRunningHosts int
+			)
+
+			numProv, err = host.CountProvisioning()
+			if err != nil {
+				j.AddError(errors.Wrap(err, "problem getting count of pending pool size"))
+				return
+			}
+			distroRunningHosts, err = host.CountRunningHosts(j.host.Distro.Id)
+			if err != nil {
+				j.AddError(errors.Wrap(err, "problem getting count of pending pool size"))
+				return
+			}
+
+			runningHosts, err = host.CountRunningHosts("")
+			if err != nil {
+				j.AddError(errors.Wrap(err, "problem getting count of pending pool size"))
+				return
+			}
+
+			if distroRunningHosts < runningHosts/100 || distroRunningHosts < j.host.Distro.HostAllocatorSettings.MinimumHosts {
```

Comment:
This seems arbitrary but I can't think of a robust way to evaluate this. Just to double check, the motivation is that if a certain distro is asking for more than 1% of all hosts, we want to control it?

Reply:
It mirrors the logic in the population function which I think we want to do.